File: ruby/Get_Balance.rb
require 'net/http'
require 'uri'
require 'json'
require 'base64'

app_id = "API_KEY"
app_sec = "API_SECRET"
app_hash = Base64.strict_encode64("#{app_id}:#{app_sec}")

base_url = "https://api-sms.4jawaly.com/api/v1/"

query = {
  "is_active" => 1,
  "order_by" => "id",
  "order_by_type" => "desc",
  "page" => 1,
  "page_size" => 10,
  "return_collection" => 1
}

url = URI.parse(base_url + "account/area/me/packages?" + URI.encode_www_form(query))

headers = {
  "Accept" => "application/json",
  "Content-Type" => "application/json",
  "Authorization" => "Basic #{app_hash}"
}

http = Net::HTTP.new(url.host, url.port)
http.use_ssl = true
request = Net::HTTP::Get.new(url.request_uri, headers)

response = http.request(request)

if response.code != '200'
  puts "Error code: #{response.code}"
  exit
end

begin
  response_json = JSON.parse(response.body)
rescue JSON::ParserError => e
  puts "Error parsing response content: #{e}"
  exit
end

if response_json['code'] == 200
  # إذا كان فيه total_balance اطبعه
  if response_json['total_balance']
    puts "Your balance: #{response_json['total_balance']}"
  elsif response_json['collection'] && !response_json['collection'].empty?
    puts "Packages collection:"
    puts JSON.pretty_generate(response_json['collection'])
  else
    puts "No balance or package data found!"
  end
else
  puts response_json['message'] || "Unknown error"
end

File: golang/Get_Balance.go
package main

import (
	"encoding/base64"
	"encoding/json"
	"fmt"
	"io/ioutil"
	"net/http"
	"strings"
)

func main() {
	appID := "API_KEY"
	appSec := "API_SECRET"
	appHash := base64.StdEncoding.EncodeToString([]byte(appID + ":" + appSec))
	baseURL := "https://api-sms.4jawaly.com/api/v1/"

	query := map[string]string{
		"is_active":          "1",
		"order_by":           "id",
		"order_by_type":      "desc",
		"page":               "1",
		"page_size":          "10",
		"return_collection":  "1",
	}
	// Build URL with query params
	var params []string
	for k, v := range query {
		params = append(params, fmt.Sprintf("%s=%s", k, v))
	}
	url := baseURL + "account/area/me/packages"
	if len(params) > 0 {
		url += "?" + strings.Join(params, "&")
	}

	headers := map[string]string{
		"Accept":        "application/json",
		"Content-Type":  "application/json",
		"Authorization": "Basic " + appHash,
	}

	client := &http.Client{}
	req, err := http.NewRequest("GET", url, nil)
	if err != nil {
		fmt.Println("Error creating request:", err)
		return
	}
	for k, v := range headers {
		req.Header.Set(k, v)
	}

	resp, err := client.Do(req)
	if err != nil {
		fmt.Println("Error executing request:", err)
		return
	}
	defer resp.Body.Close()

	if resp.StatusCode != 200 {
		fmt.Println("Error code:", resp.StatusCode)
		return
	}

	body, err := ioutil.ReadAll(resp.Body)
	if err != nil {
		fmt.Println("Error reading response body:", err)
		return
	}

	var responseJSON map[string]interface{}
	err = json.Unmarshal(body, &responseJSON)
	if err != nil {
		fmt.Println("Error unmarshalling response JSON:", err)
		return
	}

	// Safe parsing
	code, ok := responseJSON["code"].(float64)
	if !ok {
		fmt.Println("No code field in response")
		return
	}
	if int(code) == 200 {
		if balance, ok := responseJSON["total_balance"].(float64); ok {
			fmt.Printf("Your balance: %.0f\n", balance)
		} else if collection, ok := responseJSON["collection"].([]interface{}); ok {
			b, _ := json.MarshalIndent(collection, "", "  ")
			fmt.Println("Your packages collection:")
			fmt.Println(string(b))
		} else {
			fmt.Println("No total_balance or collection in response!")
		}
	} else if msg, ok := responseJSON["message"].(string); ok {
		fmt.Println(msg)
	} else {
		fmt.Println("Unexpected response:", string(body))
	}
}
